src/js/AppStore.js

```diff
@@ -0,0 +1,184 @@
+import React from 'react';
+import Modal from 'react-modal'
+import { connect } from 'react-redux';
+import store from './store';
+import { flags } from './Flags'
+
+import AppHeader from './containers/Header';
+import HScrollMenu from './HScrollMenu';
+import AppServices from './containers/AppServices';
+import ConfirmAlert from './ConfirmAlert';
+
+import BCController from './Controllers/BCController';
+import fileSystemController from './Controllers/FileSystemController'
+
+import { updateAvailableAppStoreApps, appStoreAddAppPendingSetAppProperties } from './actions';
+
+class AppStore extends React.Component {
+    constructor() {
+        super();
+
+        this.state = {
+            confirmID: null,
+            confirmApp: null
+        }
+
+        this.cancel = this.cancel.bind(this);
+        this.confirm = this.confirm.bind(this);
+        this.onSelection = this.onSelection.bind(this);
+    }
+
+    cancel() {
+        this.setState((state, props) => {
+            return {
+                confirmID: null,
+                confirmApp: null
+            }
+        });
+    }
+
+    confirm() {
+        fileSystemController.subscribeToEvent('InstallApp', (success, params) => {
+            if (!success || !params.appUrl) {
+                console.error('error encountered when installing app');
+            }
+
+            fetch(params.appUrl + 'manifest.js')
+                .then(x => x.text())
+                .then((manifestJS) => {
+                let jsonStart = manifestJS.indexOf('{');
+                let jsonEnd = manifestJS.lastIndexOf('}') + 1;
+                let manifest = JSON.parse(manifestJS.substring(jsonStart, jsonEnd))
+
+                var appProperties = {
+                    policyAppID: manifest.appId,
+                    enabled: true
+                }
+
+                let state = store.getState();
+                var appDirEntry = state.appStore.availableApps.find(x => x.policyAppID == manifest.appId);
```

Comment:
Check that appDirEntry was found

Reply:
check added in https://github.com/smartdevicelink/generic_hmi/pull/240/commits/da5e2ba127e3604518601fd85ca7e8e7b210e3b0